vyper/parser/parser.py

```diff
@@ -572,7 +572,9 @@ def parse_body(code, context):
         return parse_stmt(code, context)
     o = []
     for stmt in code:
-        o.append(parse_stmt(stmt, context))
+        lll = parse_stmt(stmt, context)
+        if lll:
```

Comment:
More of a higher level suggestion.  Return value checking is error prone (easy to forget to do).  You might look for an alternate pattern using something like `try/except`.

Reply:
agreed